documentation/modules/upgrading/con-upgrade-custom-resources.adoc

```diff
@@ -0,0 +1,107 @@
+// This module is included in the following assemblies:
+//
+// assembly-upgrade-resources.adoc
+
+[id='con-upgrade-custom-resources-{context}']
+
+= API versioning
+
+[role="_abstract"]
+Custom resources are edited and controlled using APIs added to Kubernetes by CRDs.
+Put another way, CRDs extend the Kubernetes API to allow creation of custom resources.
+CRDs are themselves resources within Kubernetes.
+They are installed in a Kubernetes cluster to define the versions of API for the custom resource.
+Each version of the custom resource API can define its own schema for that version.
+
+For release {ProductVersion}, Strimzi custom resources must specify `{ApiVersion}` as the API version.
```

Comment:
I'm not sure what this is supposed to mean. You've not explained what happens when a client uses a URL with version X and the resource is stored at version Y (!= X). For our CRDs kube will work fine it you access a v1beta2 resource though a v1beta1 URL. I think the point we're trying to make is:

1. We want people to use create `v1beta2` resources (and the procedures will cover upgrading their existing resources to `v1beta2`)
2. If people have API server clients they should be using the `v1beta2` URLs.

Reply:
The line is removed now. Do we want to expand on this still?

Reply:
![image](https://user-images.githubusercontent.com/47596553/110146341-5950be00-7dd2-11eb-962c-74725237d67b.png)

Reply:
Last sentence:

Kubernetes clients (including the Strimzi operators) access the customer resources...